crates/cheatnet/src/state.rs

```diff
@@ -158,6 +159,24 @@ pub struct CallTrace {
     pub nested_calls: Vec<Rc<RefCell<CallTrace>>>,
     pub result: CallResult,
     pub vm_trace: Option<Vec<TraceEntry>>,
+    pub node_type: NodeType,
```

Comment:
It is not the best solution, you do not encode on the type level the information that Phantom node has no interest in the any CallTrace fields. Please make CallTrace an enum instead.

Reply:
It will be easier to implement and interpret probably as well

Reply:
True, no idea why I did it this way, will change to enum